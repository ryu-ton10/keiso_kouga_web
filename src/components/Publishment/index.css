/* ========= 出版物 =========== */
.publishment {
    position: relative;
}

/* ========= 出版物の見出し =========== */
.publishment-title {
    vertical-align: middle;
    background-repeat: no-repeat;
    background-position: center 50%;
    background-size: cover;
    background-image: url("../../assets/publishment.webp");
}

.publishment-title-rgba {
    height: 100%;
    background: rgba(0, 0, 0, 0.4);
    display: flex;
    flex-direction: column;
    align-items: left;
    justify-content: center;
}

.publishment-title-jp {
    text-align: left;
    font-family: 'Hiragino Mincho Pro';
    color: white;
    padding: 0 0 0 40px;
    margin-bottom: 0;
}

.publishment-title-en {
    text-align: left;
    font-family: 'Hiragino Mincho Pro';
    color: white;
    padding: 0 0 0 40px;
    margin-bottom: 0;
}

/* ========= 見出しのレスポンスデザイン =========== */
@media (min-width: 280px) {
    .publishment-title {
        height: 100px;
    }
    .publishment-title p {
        padding: 0 0 0 15px;
    }
    .publishment-title-jp {
        font-size: 24pt;
    }
    .publishment-title-en {
        font-size: 9pt;
    }

}

@media (min-width: 360px) {
    .publishment-title {
        height: 120px;
    }
    .publishment-title p {
        padding: 0 0 0 20px;
    }
    .publishment-title-jp {
        font-size: 28pt;
    }
    .publishment-title-en {
        font-size: 11pt;
    }

}

@media (min-width: 576px) {
    .publishment-title {
        height: 150px;
    }
    .publishment-title p {
        padding: 0 0 0 30px;
    }
    .publishment-title-jp {
        font-size: 36pt;
    }
    .publishment-title-en {
        font-size: 14pt;
    }

}

@media (min-width: 960px) {
    .publishment-title {
        height: 180px;
    }
    .publishment-title p {
        padding: 0 0 0 40px;
    }
    .publishment-title-jp {
        font-size: 40pt;
    }
    .publishment-title-en {
        font-size: 16pt;
    }

}

/* ========= 同人誌詳細コンテナ =========== */
.publishment-cell {
    background-color: rgba(245, 245, 245, 1);
}

.publishment-icon {
    text-align: center;
}

.publishment-icon img {
    height: 300px;
    width: auto;
}

.publishment-descriptions {
    text-align: left;
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.publishment-descriptions p {
    margin-bottom: 0;
}

.publishment-name {
    font-family: 'Hiragino Mincho Pro';
    color: gray;
    padding: 20px 0 20px 0;
}

.publishment-detail {
    font-family: 'Hiragino Mincho Pro';
    color: gray;
}

/* ========= 同人誌詳細レスポンスクエリ =========== */
@media (min-width: 280px) {
    .publishment-cell {
        padding: 50px 30px 50px 30px;
    }
    .publishment-name {
        font-size: 18pt;
    }
    .publishment-detail {
        font-size: 9pt;
    }
    .publishment-descriptions img {
        padding: 10px 0 0 0;
    }
}

@media (min-width: 576px) {
    .publishment-cell {
        padding: 70px 70px 70px 70px;
    }
    .publishment-name {
        font-size: 24pt;
    }
    .publishment-detail {
        font-size: 9pt;
    }
    .publishment-descriptions img {
        padding: 20px 0 0 0;
    }
}

@media (min-width: 768px) {
    .publishment-cell {
        padding: 80px 80px 80px 80px;
    }
    .publishment-name {
        font-size: 28pt;
    }
    .publishment-detail {
        font-size: 12pt;
    }
    .publishment-descriptions img {
        padding: 20px 0 0 0;
    }
}

@media (min-width: 960px) {
    .publishment-cell {
        padding: 80px 100px 80px 100px;
    }
    .publishment-name {
        font-size: 30pt;
    }
    .publishment-detail {
        font-size: 14pt;
    }
    .publishment-descriptions img {
        padding: 20px 0 0 0;
    }
}
