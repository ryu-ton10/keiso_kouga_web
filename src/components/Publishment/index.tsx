import React from 'react';
import './index.css';
import { Row, Col, Image } from 'antd';

type Publishment_Image = {
    webp: string;
    jpeg: string;
}

type Book = {
    id: string;
    title: string;
    description: string;
    image: Publishment_Image;
    url: string;
}

type Props = {
    books: Book[];
}

const Publishment: React.FC<Props> = (props) => {

    const { books } = props;

    return (
        <div className="publishment">
            <div className="publishment-title">
                <div className="publishment-title-rgba">
                    <p className="publishment-title-jp">同人誌</p>
                    <p className="publishment-title-en">Publishment</p>
                </div>
            </div>
            <div className="publishment-cell">
                <Row gutter={24}>
                    {books.map((book) => {
                        const { title, description, image, url } = book;
                        return (
                        <>
                            <Col xs={24} sm={12} md={12} lg={10} className="publishment-icon">
                                <Image src={process.env.PUBLIC_URL + image.webp} fallback={process.env.PUBLIC_URL + image.jpeg} />
                            </Col>
                            <Col xs={24} sm={12} md={12} lg={14} className="publishment-descriptions">
                                <p className="publishment-name">{title}</p>
                                <p className="publishment-detail">{description}</p>
                                <a href={url} rel="noopener noreferrer" target="_blank">
                                    <img src={process.env.PUBLIC_URL + "assocbutt_or_detail.png"} alt="amazon icon"/>
                                </a>
                            </Col>
                        </>
                        );
                    })}
                </Row>
            </div>
        </div>
    );
}

export default Publishment;
