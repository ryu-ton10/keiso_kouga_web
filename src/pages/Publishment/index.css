/* ========= 出版物 =========== */
.publishment {
  background-color: rgba(43,75,123, 1);
}

/* ========= 出版物の見出し =========== */
.publishment-title {
  display: flex;
  flex-direction: column;
  align-items: left;
  justify-content: center;
}

.publishment-title-jp {
  text-align: center;
  font-family: 'Hiragino Mincho Pro';
  color: white;
  margin-bottom: 0;
}

.publishment-title-en {
  text-align: center;
  font-family: 'Hiragino Mincho Pro';
  color: white;
  margin-bottom: 0;
}

/* ========= 見出しのレスポンスデザイン =========== */
@media (min-width: 280px) {
  .publishment-title {
    height: 100px;
  }
  .publishment-title-jp {
    font-size: 24pt;
  }
  .publishment-title-en {
    font-size: 9pt;
  }
}

@media (min-width: 360px) {
  .publishment-title {
    height: 120px;
  }
  .publishment-title-jp {
    font-size: 28pt;
  }
  .publishment-title-en {
    font-size: 11pt;
  }
}

@media (min-width: 576px) {
  .publishment-title {
    height: 150px;
  }
  .publishment-title-jp {
    font-size: 36pt;
  }
  .publishment-title-en {
    font-size: 14pt;
  }
}

@media (min-width: 960px) {
  .publishment-title {
    height: 180px;
  }
  .publishment-title-jp {
    font-size: 40pt;
  }
  .publishment-title-en {
    font-size: 16pt;
  }
}

/* ========= 同人誌詳細コンテナ =========== */
.publishment-cell {
  text-align: center;
  padding-bottom: 50px;
}

.publishment-cell > h1, p {
  font-family: 'Hiragino Mincho Pro';
  color: white;
  text-align: center;
}

/* ========= 同人誌詳細レスポンスクエリ =========== */
@media (min-width: 280px) {
  .publishment-table {
    padding: 50px 30px;
  }
  .publishment-cell > picture {
    height: 300px;
    width: auto;
  }

  .publishment-cell > picture > img {
    max-width: 100%;
    width: auto;
  }
}

@media (min-width: 576px) {
  .publishment-table {
    padding: 30px 50px;
  }
  .publishment-cell > picture {
    height: 500px;
    width: auto;
  }
}

@media (min-width: 768px) {
  .publishment-table {
    display: grid;
    grid-template-columns: 1fr 1fr;
    padding: 80px;
  }

  .publishment-cell {
    padding: 20px 50px;
  }
}

@media (min-width: 960px) {
  .publishment-table {
    padding: 80px 200px
  }
}

@media (min-width: 1500px) {
  .publishment-table {
    padding: 120px 300px;
  }
}
