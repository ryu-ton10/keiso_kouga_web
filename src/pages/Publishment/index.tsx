import React from 'react';
import './index.css';
import { Row, Col, Image } from 'antd';

type Publishment_Image = {
  webp: string;
  jpeg: string;
  pre_webp: string;
  pre_jpeg: string;
}

type Book = {
  id: string;
  title: string;
  description: string;
  image: Publishment_Image;
  url: string;
}

type Props = {
  books: Book[];
}

function Publishment(props: Props) {

  const { books } = props;

  return (
    <div className="publishment">
      <div className="publishment-title">
        <p className="publishment-title-jp">同人誌</p>
        <p className="publishment-title-en">Publishment</p>
      </div>
      <div className="publishment-cell">
        <Row gutter={24}>
          {books.map((book) => {
            const { title, description, image, url } = book;
            return (
            <>
              <Col xs={24} sm={12} md={12} lg={10} className="publishment-icon">
                <Image
                  src={process.env.PUBLIC_URL + image.pre_webp}
                  fallback={process.env.PUBLIC_URL + image.pre_jpeg}
                  preview={{
                    src: `${process.env.PUBLIC_URL + image.webp}`
                  }}
                />
              </Col>
              <Col xs={24} sm={12} md={12} lg={14} className="publishment-descriptions">
                <p className="publishment-name">{title}</p>
                <p className="publishment-detail">{description}</p>
                <a href={url} rel="noopener noreferrer" target="_blank">
                  <img src={process.env.PUBLIC_URL + "assocbutt_or_detail.png"} alt="amazon icon"/>
                </a>
              </Col>
            </>
            );
          })}
        </Row>
      </div>
    </div>
  );
}

export default Publishment;
