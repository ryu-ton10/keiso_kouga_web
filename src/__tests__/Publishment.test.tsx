import renderer from 'react-test-renderer';
import Publishment from 'pages/Publishment';

const dummyBooks = [
  {
    id: '0',
    title: "sample title",
    description: "sample description",
    image: {
        webp: "sample.webp",
        jpeg: "sample.jpeg",
    },
  },
];

/* ---------------- snapshot test -------------------- */
it('コンポーネントが描画されること', () => {
  const component = renderer.create(
    <Publishment books={dummyBooks} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
