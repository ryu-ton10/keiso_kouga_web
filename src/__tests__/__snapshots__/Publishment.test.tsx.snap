// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`コンポーネントが描画されること 1`] = `
<div
  className="publishment"
>
  <div
    className="publishment-title"
  >
    <p
      className="publishment-title-jp"
    >
      同人誌
    </p>
    <p
      className="publishment-title-en"
    >
      Publishment
    </p>
  </div>
  <div
    className="publishment-table"
  >
    <div
      className="publishment-cell"
    >
      <picture>
        <source
          srcSet="sample.webp"
          type="image/webp"
        />
        <img
          aria-label="publishment-image"
          src="sample.jpeg"
        />
      </picture>
      <h1>
        sample title
      </h1>
      <p>
        sample description
      </p>
    </div>
  </div>
  <div
    className="booth-link"
  >
    <a
      href="https://keisokoga-moon.booth.pm/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        aria-label="booth-logo"
        src="booth_logo.png"
      />
    </a>
    <p>
      一部の同人誌は BOOTH でも扱っております
    </p>
  </div>
</div>
`;
